core/lxdprofile/name.go

```diff
@@ -11,3 +12,54 @@ import (
 func Name(modelName, appName string, revision int) string {
 	return fmt.Sprintf("%s-%s-%s-%d", names.Juju, modelName, appName, revision)
 }
+
+var defaultLXDProfileNames = []string{
+	"default",
+}
+
+// LXDProfileNames ensures that the LXD profile names are unique and are sorted
+// correctly. It removes certain profile names from the list, for example
+// "default" profile name will be removed.
+//
+// The function aims to preserve the order from which it got the results from.
+func LXDProfileNames(names []string) []string {
+	// ensure that the ones we have are unique
+	unique := make(map[string]int)
+	for k, v := range names {
+		if contains(defaultLXDProfileNames, v) {
```

Comment:
Let me consider this.

Reply:
I've inverted this, so we only select items that are prefixed correctly. I'm considering we should change the prefix for the lxd-profile work, so that we can then find the ones we set vs others that juju may set i.e. `juju-profile-$model-$application-$revision` or `juju-charm-profile-$model-$application-$revision`.

This isn't fool proof way if a user/operator decides to use `juju-` as a prefix for their own profiles.